### PR TITLE
[web3torrent] Add tracking calls

### PR DESCRIPTION
**Pull request by snario**

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -15938,7 +15938,7 @@ encoding@^0.1.11:
   dependencies:
     iconv-lite "~0.4.13"
 
-end-of-stream@1.4.1, end-of-stream@^1.0.0, end-of-stream@^1.1.0, end-of-stream@^1.4.1:
+end-of-stream@^1.0.0, end-of-stream@^1.1.0, end-of-stream@^1.4.1:
```
Comment (snario) [inline yarn.lock:15941]: Why does this seem to change so often

File: packages/web3torrent/src/analytics.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+import _ from 'lodash';
+export const track = (window.analytics && window.analytics.track) || _.noop;
```
Comment (kerzhner) [inline packages/web3torrent/src/analytics.ts:2]:
```suggestion
export const track = window.analytics?.track ?? _.noop;
```

File: packages/web3torrent/src/analytics.ts [hunk 3]
```diff
@@ -0,0 +1,3 @@
...
+export const identify = (window.analytics && window.analytics.identify) || _.noop;
```
Comment (kerzhner) [inline packages/web3torrent/src/analytics.ts:3]:
```suggestion
export const identify = window.analytics?.identify ?? _.noop;
```

File: packages/web3torrent/src/components/site-budget-table/SiteBudgetTable.tsx [hunk 4]
```diff
@@ -49,7 +50,13 @@ export const SiteBudgetTable: React.FC<SiteBudgetTableProps> = props => {
         <tbody>
           <tr className="budget-info">
             <td className="budget-button">
-              <button id="budget-withdraw" onClick={withdraw}>
+              <button
+                id="budget-withdraw"
+                onClick={() => {
+                  track("Clicked 'Withdraw'", {spendBudget, receiveBudget});
```
Comment (kerzhner) [inline packages/web3torrent/src/components/site-budget-table/SiteBudgetTable.tsx:56]: Should the event name be Withdraw? And maybe Button Click can be a property?

File: packages/web3torrent/src/library/web3torrent-lib.ts [hunk 5]
```diff
@@ -63,11 +64,9 @@ export default class WebTorrentPaidStreamingClient extends WebTorrent {
         'set outcomeAddress to sc-wallet web3 wallet address'
       );
       this.tracker.getAnnounceOpts = () => ({pseAccount: this.pseAccount});
-      window.analytics && // TODO: Create wrapper so we don't need to do this check
-        window.analytics.track('Enabled WebTorrentPaidStreamingClient', {
-          pseAccount: this.pseAccount,
-          outcomeAddress: this.outcomeAddress
-        });
+      identify(this.pseAccount, {
```
Comment (kerzhner) [inline packages/web3torrent/src/library/web3torrent-lib.ts:67]: Does this belong in the top level react component?

**Pull request merged**